src/WebJobs.Script/StorageProvider/HostAzureTableStorageProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using Azure.Core.Pipeline;
+using Azure.Data.Tables;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal sealed class HostAzureTableStorageProvider : IAzureTableStorageProvider
+    {
+        private readonly ILogger<HostAzureTableStorageProvider> _logger;
+        private readonly IDelegatingHandlerProvider _delegatingHandlerProvider;
+        private readonly TableServiceClientProvider _tableServiceClientProvider;
+        private readonly IConfiguration _configuration;
+
+        public HostAzureTableStorageProvider(IConfiguration configuration, ILogger<HostAzureTableStorageProvider> logger, AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder, IDelegatingHandlerProvider delegatingHandlerProvider)
```

Comment:
What is the FeatureFlagDisableMergedWebHostScriptHostConfiguration config merging logic that HostAzureBlobStorageProvider is doing (@fabiocav may know?) Wondering why it's only needed by one of these providers

Reply:
My understanding is that allows to get a merged configuration sources of the WebHost  `IConfiguration` and the `IConfiguration` at `ScriptHost/JobHost` level, and it reacts when we instantiate a new ScriptHost and loads its configuration entries.

For the `TableStorage`, since it doesn't depend on `JobHostInternalStorageOptions` and for AzureWebJobsStorage, it should come as environment variable, available in the WebHost `IConfiguration`, I made the (questionable?) decision of not merging configurations for this provider and depend on the WebHost one.

I'll let @fabiocav confirm that.